### ISSUE TITLE
GoDoc: -src

### ISSUE THREAD

**Issue by hhrutter** [fatih/vim-go#764]
Hi Fatih,

godoc comes with an option to show source code eg:
godoc -src fmt

Am I missing something here or is this not supported in vim-go?

What I am getting for 
GoDoc -src fmt 
is: No documentation found for "-src".

Is there any other way to open go library source files with vim-go?

Thx
Horst 

#### vim version: 
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jan 30 2016 10:09:26)
MacOS X (unix) version
Included patches: 1-1129
Compiled by MacPorts

#### vim-go version:
86bb44 after 1.4

#### go version:
1.6

**Comment by fatih** [comment]
Hi @hhrutter. Our `:GoDoc` is a command that displays the underlying documentation. It's using godoc under the hood, but in the future it might use something else. If you want to jump to a source code, please use `:GoDef` which will show you the source code of the underlying identifier. Thanks.

**Comment by hhrutter** [comment]
ok. godoc -src is already there and it works on the command line.
I understand that vim-go does not support this option neither does it intend so in the future.

GoDef: unfortunately does not work reliably for me.
Sometimes, especially for small, simple source files it works,
but once dealing with more elaborate source code it either is unable to locate an identifier
or it jumps somewhere that does not make sense or it does not do anything at all.

Even when obeying the mentioned limitations regarding the import section on rogpeppe/godef.
eg in some code having the line:
   ` fmt.Print("test")`
GoDef sometimes works when the cursor is located over `Print`and other times only when the cursor is over `fmt`.

This may or may not be an issue with vim-go - it's still frustrating because I really want to use vim-go but not being able to reliably show me the implementor of a function is a showstopper for me.

That being said I still like vim-go and appreciate all the hard work you put in.
Thanks.

**Comment by fatih** [comment]
Did you looked at our oracle functions? Such as :GoCallers, :GoCallees or :GoReferrers ? I quite don't get it what you are searching for but we have plenty of features to get information. Check out the docs please.

**Comment by hhrutter** [comment]
Tried them all.
Read the docs.

All I need is being able to see the source code for any function call under my cursor.

So I have a similar problem with GoCallees:
Sometimes it works sometimes it does not.
I have noticed, sometimes it thinks it is in a different line (1 line above) and therefore gets confused.
At this point it has to be related to the code I am looking at or its structure respectively.

I will further analyze and report my findings unless you tell me the reason for this behaviour may be a misconfiguration of vim because I am a vim beginner.
Thanks anyway.
